src/components/PopupWithForm.js

```diff
@@ -15,6 +17,15 @@ export default class PopupWithForm extends Popup {
     });
     return formInputs;
   }
+
+  setLoading(isLoading, loadingText = "Saving...") {
+    if (isLoading) {
+      this._submitButton.textContent = loadingText;
+    } else {
+      this._submitButton.textContent = this._submitButtonValue;
+    }
+  }
+
   setEventListeners() {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (evt) => {
```

Comment:
Please, delete `evt.target.reset();` from line 35 in `PopupWithForm`

You need to **reset** the form only in block `then` in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.